oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/PipelinedMergeSortTaskTest.java

```diff
@@ -89,27 +94,41 @@ private PipelinedMergeSortTask.Result runTest(Compression algorithm, Path... fil
         Path sortRoot = sortFolder.getRoot().toPath();
```

Comment:
Would it make sense to have a setup/teardown function for instantiating these objects? Like the root, `ScheduledExecutorService` and `MetricStatisticsProvider`? Then it would clearly signal that it is being instantiated for each test case. And we can ensure to deregister the statistics provider etc after each test case.

Reply:
I moved the executor to the top level, to be created before all tests and shutdown after the tests. There was a potential resource leak because I was not calling the shutdown method. For the `MetricStatisticsProvider`. I prefer to create it close to the code that does the testing. Almost all tests call this `runTest` method and the tests that don't do not need needs this object. So in this case, allocation of this helper object is closer to where they are used, which makes the logic simpler.